erpnext/selling/utils/order_for.py

```diff
@@ -0,0 +1,36 @@
+import frappe
+from frappe import _dict
+from erpnext.shopping_cart.cart import get_party
+
+def initialise_order_for ():
+	"""Initializes the "order for" feature which allows a backend user to use the
+	shopping cart in behalf of another user"""
+
+	# Initialize frappe.session.data.order_for
+	frappe.session.data["order_for"] = _dict(frappe.session.data.get("order_for", {}))
+	order_for = frappe.session.data.order_for
+
```

Comment:
```suggestion

	if frappe.session.user in ("Guest", "Administrator"):
		return
```
Add a check for session user first, since get_party inserts the customer and will lead to duplicate error entry for guest and admin